test/system/030-run.bats

```diff
@@ -533,8 +537,15 @@ json-file | f
 }
 
 @test "podman run with --net=host and --port prints warning" {
-    run_podman run -d --rm -p 8080 --net=host $IMAGE ls > /dev/null
```

Comment:
Quick followup on this: it turns out, `>/dev/null` is _not_ a NOP, it's worse: it causes all output from `run_podman` to vanish, which is anti-helpful when this test flakes, as it did twice in #8536! 

[remote f33](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6648492636504064/html/sys-remote-fedora-33-root-host.log.html#t--00037)

[remote ubuntu-20](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5733698962194432/html/sys-remote-ubuntu-20-root-host.log.html#t--00037)

Reply:
The intent is to remove all non-stderr output. We only want to capture the output of STDERR for this test. I believe you told me this was the best way to accomplish that?

Regarding the flake, I'll bet is the `--rm` conflicting with the test's own cleanup logic.

Reply:
Oh, that makes sense (the rm conflict). 

Re: BATS and stdout/stderr, unfortunately, there's no way to split them within a BATS `run` context (including `run_podman`). It's a BATS design tradeoff that works really well for the 95% case. The way I rewrote the test in this PR, I check for both the warning and some stdout; but do not (and cannot) check that the warning truly goes to stderr.